server/app/query/page.tsx

```diff
@@ -1,11 +1,189 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+
+import { StatusPill, RunTimePill } from "@/app/query/view/[id]/components";
+import {
+  StatusEvent,
+  RemoteServer,
+  RemoteServerNames,
+  IPARemoteServers, //hack until the queryId is stored in a DB
+  StatusEventByRemoteServer,
+  initialStatusEventByRemoteServer,
+} from "@/app/query/servers";
+import { getQueryByUUID, Query } from "@/data/query";
+
+type QueryData = {
+  statusEvent: StatusEventByRemoteServer;
+  query: Query;
+};
+type DataByQuery = {
+  [queryID: string]: QueryData;
+};
+
 export default function Page() {
+  const [queryIDs, setQueryIDs] = useState<string[]>([]);
+  const [dataByQuery, setDataByQuery] = useState<DataByQuery>({});
+
+  const updateData = (
+    query: Query,
+    remoteServer: RemoteServer,
+    statusEvent: StatusEvent,
+  ) => {
+    setDataByQuery((prev) => {
+      let _prev = prev;
+      if (!Object.hasOwn(prev, query.uuid)) {
```

Comment:
I must admit I don't really understand why it is needed. The comment below says what we are doing, but does not go into much detail why

Reply:
I can update the comment. the issue is that there are nested objects: `{queryID: {remoteServer: statusEvent}}`. If a given `queryId` isn't yet in the top level object, setting `prev[queryID][remoteServer] = newStatusEvent` doesn't work